src/java/io/bazel/rulesscala/scalac/ScalacWorker.java

```diff
@@ -99,14 +106,17 @@ public void work(String[] args) throws Exception {
 
   private static final String[] sourceExtensions = {".scala", ".java"};
 
-  private static List<File> extractSourceJars(CompileOptions opts, Path tmpParent)
+  private static List<File> extractSourceJars(CompileOptions opts, Path sources)
       throws IOException {
     List<File> sourceFiles = new ArrayList<File>();
 
     for (String jarPath : opts.sourceJars) {
       if (jarPath.length() > 0) {
-        Path tmpPath = Files.createTempDirectory(tmpParent, "tmp");
-        sourceFiles.addAll(extractJar(jarPath, tmpPath.toString(), sourceExtensions));
+        // Use full path as dest directory to handle duplicate file names
+        // TODO: might cause extremely long directory names, maybe replace with hash
```

Comment:
We should probably open an issue for this. My project is throwing file name too long errors after this commit

```
File name too long
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:91)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384)
        at java.nio.file.Files.createDirectory(Files.java:674)
        at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781)
        at java.nio.file.Files.createDirectories(Files.java:767)
        at io.bazel.rulesscala.scalac.ScalacWorker.extractSourceJars(ScalacWorker.java:118)
        at io.bazel.rulesscala.scalac.ScalacWorker.work(ScalacWorker.java:37)
        at io.bazel.rulesscala.worker.Worker.persistentWorkerMain(Worker.java:84)
        at io.bazel.rulesscala.worker.Worker.workerMain(Worker.java:39)
        at io.bazel.rulesscala.scalac.ScalacWorker.main(ScalacWorker.java:25)
```

Reply:
@caseyduquettesc I've proposed a change to address that https://github.com/bazelbuild/rules_scala/pull/1323 Sorry for inconvenience.

Reply:
Not a problem! Thanks for the quick response!